### PR TITLE
Config changes

### PR DESCRIPTION
**Pull request by Allcharles**
# Modify Config Service

Updated the apps config/environment service

## Changes

- Rename from `AppConfigService` to `ConfigService`
- Added `links` subsection to config
- Moved google maps key into `environment` subsection

## Problems

- This will require a change to the deployment process (will be done after approval)

## Final Checklist

- [x] Assign reviewers if you have permission
- [x] Assign labels if you have permission
- [x] Link issues related to PR
- [x] Ensure project linter is not producing any warnings (`npm run lint`)
- [x] Ensure build is passing on all browsers (`npm run test:all`)
- [ ] Ensure CI build is passing
- [ ] Ensure docker container is passing ([docs](https://github.com/QutEcoacoustics/workbench-client#docker))

### CODE REVIEW

File: src/app/helpers/app-initializer/app-initializer.ts [hunk 1]
```diff
@@ -56,9 +54,12 @@ export interface Environment {
...
   siteDir: string;
-  ga: {
-    domain: string;
-    trackingId: string;
+  keys: {
+    googleMaps: string;
+    googleAnalytics: {
+      domain: string;
+      trackingId: string;
+    };
```
Comment (atruskie) [inline src/app/helpers/app-initializer/app-initializer.ts:64]: deploy breaking

File: src/app/helpers/app-initializer/app-initializer.ts [hunk 2]
```diff
@@ -38,13 +38,11 @@ export class AppInitializer {
...
  */
 export interface Values {
-  keys: {
-    googleMaps: string;
-  };
   brand: {
     name: string;
     title: string;
   };
+  links: Record<string, string>;
```
Comment (atruskie) [inline src/app/helpers/app-initializer/app-initializer.ts]: I don't like that this is obscured/dynamic.

This interface should be a static specification for a config file.

File: src/app/helpers/app-initializer/app-initializer.ts [hunk 3]
```diff
@@ -38,13 +38,11 @@ export class AppInitializer {
  * App values
  */
 export interface Values {
```
Comment (atruskie) [inline src/app/helpers/app-initializer/app-initializer.ts:40]: Why do we have two interfaces?

File: src/assets/environment.json [hunk 4]
```diff
@@ -4,19 +4,22 @@
...
   "values": {
-    "keys": {
-      "googleMaps": ""
-    },
     "brand": {
       "name": "<< brandName >>",
       "title": "<< brandName >> | Acoustic Workbench"
     },
+    "links": {
+      "githubBawServerIssues": "https://github.com/QutEcoacoustics/baw-server/issues"
```
Comment (atruskie) [inline src/assets/environment.json]: dislike name. 

suggest: `sourceRepository`

File: src/app/helpers/app-initializer/app-initializer.ts (hunk 3 above)
Comment (atruskie) [reply src/app/helpers/app-initializer/app-initializer.ts:40]: Maybe I'm blind but I can't see where the `Values` interface is used within the `Environment` interface

**Review by atruskie — COMMENTED** [review]
This is a blocking change and won't be approved until the requisite change is made to deploy scripts and config template in baw-private

File: src/app/helpers/app-initializer/app-initializer.ts (hunk 3 above)
Comment (Allcharles) [reply src/app/helpers/app-initializer/app-initializer.ts:40]: Its used in the `Configuration` interface

**Review by atruskie — APPROVED** [review]

**Pull request merged**